test/tests/functional/pbs_reservations.py

```diff
@@ -1487,45 +1475,58 @@ def test_job_array_within_recurring_reservation(self):
                            id=jids[1], extend='t')
         self.server.expect(JOB, {'job_state=Q': 3}, count=True,
                            id=jids[2], extend='t')
-        # At end of first instance of resrvation  ,verify running sub-job
-        # terminated
+        # At end of first instance of reservation ,verify running sub-job
+        # should be finished
         self.logger.info(
-            'Waiting 60 seconds for first  instance of reservation to end')
-        time.sleep(62)
-        job_list = subjid
-        job_list.pop()
-        job_list.pop()
-        for subjob in job_list:
-            self.server.expect(JOB, {'job_state': 'F', 'substate': '93'},
-                               extend='x', id=subjob)
+            'Waiting 30 sec job-array 1 and 2 to be finished')
+        self.server.expect(JOB, {'job_state=F': 3}, extend='xt',
```

Comment:
You have already checked that reservation is running, you can reduce the offset to 20 or less